File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task1-Server/src/main/java/com/pet/hotel/data/database/entities/RentEntity.java
package com.pet.hotel.data.database.entities;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.stereotype.Component;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "rent")
@NoArgsConstructor
@Component
public class RentEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "rent_id")
    @Setter(AccessLevel.PUBLIC)
    private int rentId;

    @Column(name = "begin_date", nullable = false)
    private LocalDateTime beginDate;

    @Column(name = "end_date", nullable = false)
    private LocalDateTime endDate;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "room_id", nullable = false)
    private RoomEntity room;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "pet_id")
    private PetEntity pet;

    @OneToMany(fetch = FetchType.EAGER, mappedBy = "rent")
    private List<DietEntity> diets;

    public RentEntity(LocalDateTime beginDate, LocalDateTime endDate, RoomEntity room, PetEntity pet, List<DietEntity> diets) {
        this.beginDate = beginDate;
        this.endDate = endDate;
        this.room = room;
        this.pet = pet;
        this.diets = diets;
    }
}

File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task4-MobileClient/src/main/java/com/animal/hotel/data/user/entities/RentEntity.kt
package com.animal.hotel.data.user.entities

import com.animal.hotel.domain.models.user.Rent
import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class RentEntity(
    val rentId: Int,
    val beginDate: LocalDateTime,
    val endDate: LocalDateTime,
    val room: RoomEntity,
    val pet: PetEntity
) {
    fun toRent(): Rent {
        return  Rent(rentId, beginDate, endDate, room.toRoom(), pet.toPet())
    }

    companion object {
        fun toRentEntity(rent: Rent) =  RentEntity (
            rent.rentId,
            rent.beginDate,
            rent.endDate,
            RoomEntity.toRoomEntity(rent.room),
            PetEntity.toPetEntity(rent.pet)
        )
    }
}
